### PR TITLE
fix vertical line spectrum plot bug

### PR DESCRIPTION
**Pull request by to266**

### CODE REVIEW

File: hyperspy/drawing/spectrum.py [hunk 1]
```diff
@@ -332,19 +332,18 @@ def update(self, force_replot=False):
...
-        new_axis = self.axes_manager.signal_axes[0].axis
-        if len(new_axis) != len(self.axis) or \
-           (len(new_axis) == len(self.axis) and
-                np.any(np.not_equal(new_axis, self.axis))):
-            self.axis = new_axis
-            self.ax.set_xlim(self.axis[0], self.axis[-1])
-            self.line.set_data(self.axis, ydata)
+        old_xaxis = self.line.get_xdata()
+        if len(old_xaxis) != len(self.axis.axis) or \
+           (len(old_xaxis) == len(self.axis.axis) and
```
Comment (vidartf) [inline hyperspy/drawing/spectrum.py]: `len(self.axis.axis` --> `self.axis.size` ?

Comment (to266) [inline hyperspy/drawing/spectrum.py]: Done, thanks

File: hyperspy/drawing/spectrum.py [hunk 2]
```diff
@@ -332,19 +332,18 @@ def update(self, force_replot=False):
...
-        new_axis = self.axes_manager.signal_axes[0].axis
-        if len(new_axis) != len(self.axis) or \
-           (len(new_axis) == len(self.axis) and
-                np.any(np.not_equal(new_axis, self.axis))):
-            self.axis = new_axis
-            self.ax.set_xlim(self.axis[0], self.axis[-1])
-            self.line.set_data(self.axis, ydata)
+        old_xaxis = self.line.get_xdata()
+        if len(old_xaxis) != self.axis.size or \
+           (len(old_xaxis) == self.axis.size and
```
Comment (vidartf) [inline hyperspy/drawing/spectrum.py]: `len(old_xaxis) != self.axis.size or (len(old_xaxis) == self.axis.size and <something>`
Second size comparison (`==`) is redundant, right? Or am I missing something in the logic?

**Comment by vidartf** [comment]
Surprisingly little abuse of the direct `axis` other places in the code (nothing as far as I could find). Makes you wonder why it was like that in the first place.

File: hyperspy/drawing/spectrum.py (hunk 2 above)
Comment (to266) [inline hyperspy/drawing/spectrum.py]: yeah it is. Sorry that I'm doing this in a hurry. And thanks again

**Comment by to266** [comment]
Yeah I was ready to fix a lot more, however very few places needed touching... I'll have to keep it in mind in case image plotting requires something similar (it may be using axes (not arrays) already, don't know..)

**Pull request merged**